core/src/main/java/io/grpc/internal/AbstractChannelBuilder.java

```diff
@@ -45,7 +48,8 @@
  *
  * @param <BuilderT> The concrete type of this builder.
  */
-public abstract class AbstractChannelBuilder<BuilderT extends AbstractChannelBuilder<BuilderT>> {
+public abstract class AbstractChannelBuilder<BuilderT extends AbstractChannelBuilder<BuilderT>>
+    extends ChannelBuilder<BuilderT> {
```

Comment:
Based on names, this is a very confusing type hierarchy.

Could the `AndroidChannelBuilder` just extends `OkHttpChannelBuilder`, and then not need a new abstract interface-like supertype?